Add warning about bug in some emacs versions

ian-kelling opened the pull request:
#11 is fixed upstream. Add warning for affected versions and call it fixed.

ian-kelling committed:
Add warning about bug in some emacs versions

monsanto commented:
Could you also add the warning to README.md? Otherwise looks good to me

monsanto commented on readline-complete.el:
```diff
@@ -135,10 +135,18 @@
 ;; editline instead of readline. Consider recompiling bash, python,
 ;; etc with readline support instead.
 
+;; *** Note: emacs contained a bug from ~24.3.91 to ~24.5-rc3-fixed where
```
Your bug was reported for 25.0.50, so the bit about 24.5-rc3-fixed seems off to me? Or do you mean that is when the bug was introduced?

monsanto commented on readline-complete.el:
```diff
@@ -135,10 +135,18 @@
 ;;; Code:
 
+(when (or (and (= 24 emacs-major-version) (>= 3 emacs-minor-version))
+          (and (= 25 emacs-major-version) (<= 0 emacs-minor-version)))
```
Unless this is going in `25.1`, you may want to extract the exact patch version from `emacs-version` instead of using `emacs-minor-version`. Otherwise someone updating to the latest may be confused that they still get the warning.

For reference, here is how `emacs-*-version` are defined:

```elisp
(defconst emacs-major-version
  (progn (string-match "^[0-9]+" emacs-version)
         (string-to-number (match-string 0 emacs-version)))
  "Major version number of this version of Emacs.
This variable first existed in version 19.23.")

(defconst emacs-minor-version
  (progn (string-match "^[0-9]+\\.\\([0-9]+\\)" emacs-version)
         (string-to-number (match-string 1 emacs-version)))
  "Minor version number of this version of Emacs.
This variable first existed in version 19.23.")
```

monsanto replied:
hmm after thinking about it a bit more this is probably more trouble than its worth. I personally run Emacs HEAD but others probably don't. I'll leave it up to you

ian-kelling commented:
I think I was a bit tired, didn't fully think this through. I've updated it to use patch version, and got the right version.

ian-kelling commented:
... now.

ian-kelling commented:
Addressed your comments, except, normalizing patch-ver to 0 won't work. It would include 0 when I test <= and exclude when I test >=, but I'd like to include it on both conditions. I couldn't think of a better way to do it, without making things overly complicated / verbose.

monsanto commented:
I think you are mistaken:

```elisp
(if patch-ver (>= 91 patch-ver) t) -->
(if nil (>= 91 nil) t) -->
t

(>= 91 patch-ver) -->
(>= 91 0) -->
t
```

and

```elisp
(if patch-ver (<= patch-ver 50) t) -->
(if nil (<= nil 50) t) -->
t

(<= patch-ver 50) -->
(<= 0 50) -->
t
```

ian-kelling commented:
Heh, we were both right.

    (>= 91 patch-ver)
should be
    (>= patch-ver 91)

However, I think you were right to say it's not worth it. I thought, I bet someone else did this already, and they did:  http://www.emacswiki.org/emacs/versions.el, and if it's not worth including a library (which I don't think it is), it's not worth having this complexity in our code. I've updated the pull request to be just the comment.

monsanto commented:
Fair enough. Just add it to the README as well per my first comment and I will merge :)

monsanto commented:
you alive @ian-kelling? i'll amend your commit myself if you don't do it by next time i check in.
